snapcraft/internal/build_providers/_multipass/_windows.py

```diff
@@ -0,0 +1,197 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import os.path
+import requests
+import shutil
+import simplejson
+import subprocess
+import sys
+import tempfile
+
+from snapcraft.file_utils import calculate_sha3_384
+from snapcraft.internal.build_providers.errors import (
+    ProviderMultipassDownloadFailed,
+    ProviderMultipassInstallationFailed,
+)
+from snapcraft.internal.indicators import download_requests_stream
+
+if sys.platform == "win32":
+    import winreg
+
+
+logger = logging.getLogger(__name__)
+
+
+_MULTIPASS_RELEASES_API_URL = (
+    "https://api.github.com/repos/CanonicalLtd/multipass/releases/latest"
+)
+_MULTIPASS_DL_VERSION = "0.8.0"
+_MULTIPASS_DL_NAME = "multipass-{version}+win-win64.exe".format(
+    version=_MULTIPASS_DL_VERSION
+)
+_MULTIPASS_DL_SHA3_384 = "a1ac2eeb77b2a98fe5dee198be70dbf1a985d94b9707ce33ea0d3828dbc90d07fccb9662b7c97a3cfa194895b4f56676"
```

Comment:
I must say I dislike this approach, as it will mean that as soon as people install this, they'll be greeted with a notification that their Multipass is old…

If you're worried about us adding some other files there, we can certainly discuss how to ensure that doesn't break you.

FWIW we also plan to add Multipass to `choco` (CanonicalLtd/multipass#826), would that be better for you guys?

Reply:
@cjp256 ACK, I didn't read enough of the code to realize you're falling back to pointing the user at the website.

As you were, then ;)